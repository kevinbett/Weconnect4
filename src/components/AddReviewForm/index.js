import React from 'react';
import api  from '../../apiServices/api';
import './style.css';
import swal from 'sweetalert';
import Navigation from '../Navigation';

export const validate = (data) => {
  const errors = {};
  return errors;
 };

class AddReviewForm extends React.Component {
  constructor(props) {
    super(props); 
    this.state = {
      message: undefined,
      data: {
        feedback:'',
        id: ''
  
      }
    }
  
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  
  componentDidMount() {
    if (!this.props) {
      window.location.assign('/viewbusiness');
    }


    const data = this.props.location.query;

    this.setState({
        data: {
          ...this.state.data,
          id: data.review.id
        }
      });

  }

  onChange(e) {
    this.setState({ data: { ...this.state.data, [e.target.name]: e.target.value } });
  }

  onSubmit = (e) => {
    e.preventDefault();
    const errors = validate(this.state.data);
    this.setState({ errors });
    if (Object.keys(errors).length === 0) {
      this.setState({ loading: true });
      api.business.review(this.state.data.id, this.state.data).then(() => {
      this.props.history.push(`/profile/${this.state.data.id}`);
        swal('review added succesfully', 'success')
      })
        .catch(err => {
          swal({title:err.response.data.message})
          this.setState({ errors: err.response, loading: false });
          this.props.history.push(`/profile/${this.state.data.id}`)
        });
    }
  };

  
  render() {
    return (
        <div>
        <Navigation/>
    <div id="main-registration-container">
     <div id="register">
        <h3>Enter your Review</h3>
        <form onSubmit={this.onSubmit}>
        <p className="errorMsg">{this.state.message}</p>
        <label>Review</label>
        <input type="text"  onChange={this.onChange} value={this.state.data.feedback} name="feedback"  />

        <input type="submit" className="button"  value="Submit Review"/>
        </form>
    </div>
</div>
</div>

      );
  }


}


export default AddReviewForm;