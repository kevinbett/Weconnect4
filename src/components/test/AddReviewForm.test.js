import AddReviewForm from "../AddReviewForm/index"
import { shallow } from "enzyme"

describe('Add Review Page Renders',() => {
    it ('populates add review form', () => { 
        const location = {
            query: {
                review: {
                    category: "fast food",
                    id: 122,
                    location: "roysambu",
                    name: "minenetoo",
                    type:"cuisine",
                    user_id:1
                }
            }
        }
        expect(shallow(<AddReviewForm location={location}/>)).toMatchSnapshot()
    });

} )