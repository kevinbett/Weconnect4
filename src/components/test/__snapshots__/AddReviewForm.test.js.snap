// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Review Page Renders populates add review form 1`] = `
<div>
  <Navigation />
  <div
    id="main-registration-container"
  >
    <div
      id="register"
    >
      <h3>
        Enter your Review
      </h3>
      <form
        onSubmit={[Function]}
      >
        <p
          className="errorMsg"
        />
        <label>
          Review
        </label>
        <input
          name="feedback"
          onChange={[Function]}
          type="text"
          value=""
        />
        <input
          className="button"
          type="submit"
          value="Submit Review"
        />
      </form>
    </div>
  </div>
</div>
`;
